### PR TITLE
Ticket Validation - Add missing logs.

### PR DESCRIPTION
**Pull request by ArturFriesen**
## Description
Add missing logs. Change some others.

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-10740

### CODE REVIEW

**Review by naveeddotio — APPROVED** [review]

File: src/xcode/ENA/ENASecurity/Sources/ENASecurity/TrustEvaluation.swift [hunk 1]
```diff
@@ -18,20 +18,23 @@ public class TrustEvaluation {
...
 
-            return check(serverKeyData: serverCertificateData, against: jwkSet)
+            return check(serverKeyData: serverCertificateData, against: jwkSet, logMessage: logMessage)
         } else {
             return .failure(.CERT_CHAIN_EMTPY)
         }
     }
 
-    public func check(serverKeyData: Data, against jwkSet: [JSONWebKey]) -> Result<Void, TrustEvaluationError> {
+    public func check(serverKeyData: Data, against jwkSet: [JSONWebKey], logMessage: ((String) -> Void)?) -> Result<Void, TrustEvaluationError> {
```
Comment (marcussc) [inline src/xcode/ENA/ENASecurity/Sources/ENASecurity/TrustEvaluation.swift:32]: Nice idea!

File: src/xcode/ENA/ENA/Source/Services/TicketValidation/ValidationDecorator/TicketValidationDecoratorIdentityDocumentProcessor.swift [hunk 2]
```diff
@@ -11,6 +11,8 @@ final class TicketValidationDecoratorIdentityDocumentProcessor {
 		completion:
 		@escaping (Result<TicketValidationServiceIdentityDocumentValidationDecorator, ServiceIdentityValidationDecoratorError>) -> Void
 	) {
+		Log.info("Validate identity document.", log: .ticketValidation)
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Services/TicketValidation/ValidationDecorator/TicketValidationDecoratorIdentityDocumentProcessor.swift:14]:
```suggestion
		Log.info("Start validation of identity document.", log: .ticketValidation)
```

File: src/xcode/ENA/ENA/Source/Services/TicketValidation/TicketValidation.swift [hunk 3]
```diff
@@ -203,6 +211,8 @@ final class TicketValidation: TicketValidating {
 					completion(result)
 				}
 			case .failure(let error):
+				Log.error("Failed to request document of validation decorator with error: \(error)", log: .ticketValidation)
```
Comment (marcussc) [inline src/xcode/ENA/ENA/Source/Services/TicketValidation/TicketValidation.swift:214]: Are we sure the error does not contain any private information?

**Review by marcussc — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Services/TicketValidation/TicketValidation.swift (hunk 3 above)
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Services/TicketValidation/TicketValidation.swift:214]: I hope that there is common consent to not put any private data into errors.
To be safe I looked into the ServiceError and the custom errors. Didn't see anything private.

**Pull request merged**